packages/kbn-search-index-documents/components/documents_overview.tsx

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import {
+  EuiFieldSearch,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPanel,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { ChangeEvent } from 'react';
+import { IngestionMethod } from '../types';
+
+interface DocumentsProps {
+  accessControlSwitch?: React.ReactNode;
+  documentComponent: React.ReactNode;
+  ingestionMethod: IngestionMethod;
+  searchQueryCallback: (searchQuery: string) => void;
+}
+export const DocumentsOverview: React.FC<DocumentsProps> = ({
+  accessControlSwitch,
+  documentComponent,
+  ingestionMethod,
+  searchQueryCallback,
+}) => {
+  return (
+    <EuiPanel hasBorder={false} hasShadow={false} paddingSize="none">
+      <EuiSpacer />
+      <EuiFlexGroup direction="column">
+        <EuiFlexItem>
+          <EuiFlexGroup direction="row" alignItems="center">
+            <EuiFlexItem className="enterpriseSearchDocumentsHeader" grow={false}>
+              <EuiTitle>
+                <h2>
+                  {i18n.translate('searchIndexDocuments.documents.title', {
+                    defaultMessage: 'Browse documents',
+                  })}
+                </h2>
+              </EuiTitle>
+            </EuiFlexItem>
+            {accessControlSwitch}
+            <EuiFlexItem>
+              <EuiFieldSearch
+                data-telemetry-id={`entSearchContent-${ingestionMethod}-documents-searchDocuments`}
```

Comment:
we might want to make this telemetry id a prop so that we can customize it between dedicated and serverless.

Also if `ingestionMethod` is only used for `data-telemetry-id` then we should abstract that away and not define `IngestionMethod` in this package.

We could possible have a single `data-telemetry-id-prefix` prop on the `DocumentList` or `DocumentOverview` and then use that in a string template to make unique `data-telemetry-id`s for multiple components if we have several of them.

In this case it would be something like:
```suggestion
  data-telemetry-id={`${dataTelemetryIdPrefix}-documents-searchDocuments`}
```

and in the plugin we would do 
```
<DocumentOverview dataTelemetryIdPrefix={`entSearchContent-${ingestionMethod}`} ... />
```